agent/api/task/task_linux.go

```diff
@@ -355,3 +357,37 @@ func (task *Task) BuildCNIConfig(includeIPAMConfig bool, cniConfig *ecscni.Confi
 
 	return cniConfig, nil
 }
+
+// BuildCNIConfigBridgeMode builds a list of CNI network configurations for a task in docker bridge mode.
+// Currently the only plugin in available is for Service Connect
+func (task *Task) BuildCNIConfigBridgeMode(cniConfig *ecscni.Config, containerName string) (*ecscni.Config, error) {
+	if !task.IsNetworkModeBridge() || !task.IsServiceConnectEnabled() {
+		return nil, errors.New("only bridge-mode Service-Connect-enabled task should invoke BuildCNIConfigBridgeMode")
+	}
+
+	var netconf *libcni.NetworkConfig
+	var ifName string
+	var err error
+
+	scPauseIPConfig := task.GetServiceConnectRuntimeConfig().PauseContainerIPConfig
```

Comment:
Same Q asked in https://github.com/aws/amazon-ecs-agent/pull/3265/files#r914210160. Thanks!

Reply:
it won't be null if we already get here, but it might be null when we populate pause container IPs in `task.PopulateServiceConnectPauseIPConfig` - I'll make sure to create `RuntimeConfig` if it does not already exist

Reply:
Created a new field `NetworkConfig` that's no longer nested in `RuntimeConfig` to avoid the complication that multiple containers could potentially be creating it.